ClickableTransparentOverlay/Overlay.cs

```diff
@@ -14,75 +15,113 @@ namespace ClickableTransparentOverlay
+        private Vector4 clearColor;
+        private Dictionary<string, Texture> loadedImages;
         private static bool terminal = true;
+        
+        private Thread renderThread;
+        private volatile CancellationTokenSource cancellationTokenSource;
+        private volatile bool overlayIsReady;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Overlay"/> class.
         /// </summary>
-        static Overlay()
+        public Overlay()
         {
             clearColor = new Vector4(0.00f, 0.00f, 0.00f, 0.00f);
             loadedImages = new Dictionary<string, Texture>();
-            window = new Sdl2Window(
-                "Overlay",
-                0,
-                0,
-                2560,
-                1440,
-                SDL_WindowFlags.Borderless |
+        }
+
+        /// <summary>
+        /// Starts the overlay
+        /// </summary>
+        /// <returns>Task that finishes once the overlay is ready</returns>
+        public async Task Start()
+        {
+            cancellationTokenSource = new CancellationTokenSource();
+            
+            renderThread = new Thread(async () =>
+            {
+                window = new Sdl2Window(
+                    "Overlay",
+                    0,
+                    0,
+                    2560,
+                    1440,
+                    SDL_WindowFlags.Borderless |
                     SDL_WindowFlags.AlwaysOnTop |
                     SDL_WindowFlags.SkipTaskbar,
-                false);
-            graphicsDevice = VeldridStartup.CreateDefaultD3D11GraphicsDevice(
-                new GraphicsDeviceOptions(false, null, true),
-                window);
-            commandList = graphicsDevice.ResourceFactory.CreateCommandList();
-            imController = new ImGuiController(
-                graphicsDevice,
-                graphicsDevice.MainSwapchain.Framebuffer.OutputDescription,
-                window.Width,
-                window.Height);
-            window.Resized += () =>
+                    false);
+                graphicsDevice = VeldridStartup.CreateDefaultD3D11GraphicsDevice(
+                    new GraphicsDeviceOptions(false, null, true),
+                    window);
+                commandList = graphicsDevice.ResourceFactory.CreateCommandList();
+                imController = new ImGuiController(
+                    graphicsDevice,
+                    graphicsDevice.MainSwapchain.Framebuffer.OutputDescription,
+                    window.Width,
+                    window.Height);
+                window.Resized += () =>
+                {
+                    graphicsDevice.MainSwapchain.Resize((uint)window.Width, (uint)window.Height);
+                    imController.WindowResized(window.Width, window.Height);
+                };
+                
+                NativeMethods.InitTransparency(window.Handle);
+                NativeMethods.SetOverlayClickable(window.Handle, false);
```

Comment:
I wonder why things don't work if we do line:51 - line:77 before creating/starting the Render thread.

Reply:
Veldrid spawns the Sdl2Window on the same Thread the Sdl2Window.ctor is called on. And I guess once you click in the overlay something happens that causes the Main Thread to deadlock

In theory instead of using my MR you could have also set the flag `threadedProcessing` (Sdl2Window.ctor) to true. But the way they implemented the threading in veldrid causes my cpu to spike to ~16% usage vs the ~2% in the MR solution. (There is an issue for that on their repo #mellinoe/veldrid/issues/216)